### DISCUSSION TITLE
Link preview image not working in Mastodon - any ideas?

### DISCUSSION THREAD

**Discussion by danbraun** [11ty/eleventy#2788] [Q&A]
Hello, I'm sharing a link to my Eleventy blog, but the image doesn't show on Mastodon.
<img width="510" alt="image" src="https://user-images.githubusercontent.com/6805888/216849700-b5671518-ccb0-4095-9f22-9e42127e3ba6.png">
I have the [requisite tags](https://blog.danbraun.me/posts/drone-music/) including `<meta property="og:image" content="https://blog.danbraun.me/assets/social.png">` and `<meta name="twitter:image" content="https://blog.danbraun.me/assets/social.png">`
It seems to work in online inspectors and LinkedIn (I don't have other socials to test).
Also, I'm using the eleventy-plugin-metagen plugin.

**Comment by pdehaan** [comment]
Not sure this would be an Eleventy issue. You might need to post in `eleventy-plugin-metagen` repo; although both the `og:image` and `twitter:image` assets loaded as expected when I opened them in a new tab.

I also found https://github.com/mastodon/mastodon/issues/13213 which might be relevant:
> _"Sometimes the preview isn't ready by the time the toot goes through the streaming API, but it's there later."_

I can't remember if that was a Twitter/OpenGraph issue also, if it's the first time encountering that exact URL, it might not have scraped+processed the metadata yet, but image previews might be available 1-5 minutes later. :shrug:
